bitsail-connectors/connector-rocketmq/src/main/java/com/bytedance/bitsail/connector/rocketmq/sink/RocketMQProducer.java

```diff
@@ -0,0 +1,255 @@
+  private final boolean batchFlushEnable;
+  private final boolean logFailuresOnly;
+  private final boolean enableSyncSend;
+
+  /* acl control */
+  private final String accessKey;
+  private final String secretKey;
+
+  /* props for producer */
+  private final int failureRetryTimes;
+  private final int sendMsgTimeout;
+  private final int maxMessageSize;
+
+  /* messages and producer */
+  private final List<Message> messageList;
+  private transient DefaultMQProducer producer;
+
+  private transient SendCallback callBack;
+  private transient volatile Exception flushException;
+
+  @Setter
+  private boolean enableQueueSelector;
+  private transient HashQueueSelector queueSelector;
+
+  public RocketMQProducer(RocketMQSinkConfig sinkConfig) {
+    this.nameServerAddress = sinkConfig.getNameServerAddress();
+    this.producerGroup = sinkConfig.getProducerGroup();
+    this.accessKey = sinkConfig.getAccessKey();
+    this.secretKey = sinkConfig.getSecretKey();
+    this.failureRetryTimes = sinkConfig.getFailureRetryTimes();
+    this.sendMsgTimeout = sinkConfig.getSendMsgTimeout();
+    this.maxMessageSize = sinkConfig.getMaxMessageSize();
+
+    this.batchSize = sinkConfig.getBatchSize();
+    this.batchFlushEnable = sinkConfig.isEnableBatchFlush();
+    this.logFailuresOnly = sinkConfig.isLogFailuresOnly();
+    this.enableSyncSend = sinkConfig.isEnableSyncSend();
+
+    this.enableQueueSelector = false;
+    this.messageList = new ArrayList<>();
+  }
+
+  /**
+   * initialize message list, open rocketmq producer
+   */
+  public void open() {
+    // step1: initialize callback and queue-selector
+    if (logFailuresOnly) {
+      callBack = new SendCallback() {
+        @Override
+        public void onSuccess(SendResult sendResult) {}
+
+        @Override
+        public void onException(Throwable throwable) {
+          LOG.error("Error while sending record to rocketmq: " + throwable.getMessage(), throwable);
+        }
+      };
+    } else {
+      callBack = new SendCallback() {
+        @Override
+        public void onSuccess(SendResult result) {}
+
+        @Override
+        public void onException(Throwable e) {
+          if (flushException == null) {
+            flushException = new IOException(e.getMessage(), e);
+          }
+        }
+      };
+    }
+
+    if (enableQueueSelector) {
+      this.queueSelector = new HashQueueSelector();
+    }
+
+    // step2: open a producer
+    if (!StringUtils.isEmpty(accessKey) && !StringUtils.isEmpty(secretKey)) {
+      AclClientRPCHook rpcHook = new AclClientRPCHook(new SessionCredentials(accessKey, secretKey));
+      this.producer = new DefaultMQProducer(producerGroup, rpcHook);
+    } else {
+      this.producer = new DefaultMQProducer(producerGroup);
+    }
+
+    producer.setNamesrvAddr(nameServerAddress);
+    producer.setRetryTimesWhenSendFailed(failureRetryTimes);
+    producer.setSendMsgTimeout(sendMsgTimeout);
+    producer.setMaxMessageSize(maxMessageSize);
```

Comment:
can we pass some extra properties to the producer? For example we can pass a `Map<String,String>` as a user defined config and add it to the producer.